src/escpos/printer.py

```diff
@@ -52,47 +54,38 @@ def __init__(self, idVendor, idProduct, timeout=0, in_ep=0x82, out_ep=0x01, *arg
 
     def open(self):
         """ Search device on USB tree and set it as escpos device """
-        self.device = usb.core.find(idVendor=self.idVendor, idProduct=self.idProduct)
-        if self.device is None:
+        self.ctx = USBContext()
+        dev = self.ctx.getByVendorIDAndProductID(self.idVendor, self.idProduct)
+        if dev is None:
             raise USBNotFoundError("Device not found or cable not plugged in.")
 
-        check_driver = None
-
-        try:
-            check_driver = self.device.is_kernel_driver_active(0)
-        except NotImplementedError:
-            pass
+        self.device = dev.open()
+        if platform.system() == 'Linux' and self.device.kernelDriverActive(self.INTERFACE):
+            self.device.detachKernelDriver(self.INTERFACE)
 
-        if check_driver is None or check_driver:
-            try:
-                self.device.detach_kernel_driver(0)
-            except usb.core.USBError as e:
```

Comment:
Would it be possible to also check for this exception in your code?

Reply:
I am not sure if this is necesary with libusb1.